sopel/coretasks.py

```diff
@@ -741,7 +742,17 @@ def receive_cap_ack_sasl(bot):
         mech = bot.config.core.server_auth_sasl_mech
     if not password:
         return
+
     mech = mech or 'PLAIN'
+    available_mechs = bot.server_capabilities.get('sasl', '')
+    available_mechs = available_mechs.split(',') if available_mechs else []
+
+    if available_mechs and mech not in available_mechs:
```

Comment:
Why the check for `available_mechs in available_mechs`? Am I missing something here?

Reply:
It's grouped like `if (available_mechs) and (mech not in available_mechs)`; see the comment below for why.